src/pages/Profiles.vue

```diff
@@ -478,22 +479,30 @@ export default class Profiles extends Vue {
         });
     }
 
-    importProfileUsingCode() {
-        Axios.get(`https://r2modman-hastebin.herokuapp.com/raw/${this.profileImportCode}`)
-            .then(resp => resp.data)
-            .then(async resp => {
-                if (resp.startsWith("#r2modman")) {
-                    const buf = Buffer.from(resp.substring(9).trim(), 'base64');
-                    await FileUtils.ensureDirectory(path.join(PathResolver.ROOT, '_import_cache'));
-                    await fs.writeFile(path.join(PathResolver.ROOT, '_import_cache', 'import.r2z'), buf);
-                    await this.importProfileHandler([path.join(PathResolver.ROOT, '_import_cache', 'import.r2z')]);
-                } else {
-                        throw new Error('Code invalid, no profile is associated with this code');
-                }
-            }).catch(e => {
+    async parseProfileResponse(response_data: string) {
```

Comment:
With a quick glance, seems like doing this would mean splitting right about the whole file down into separate modules. I'll leave it out of this PR, but I'll drag it through the post-mortem and wrap-up so we'll have task for refactoring the mod manager code into the modules; so that it can be used more easily in the future.

Reply:
Opened #883 regarding this